Homework/Van_Le/lesson4/ProductArrayExceptSelf/ProductArrayExceptSelf.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Linq;
+
+namespace ProductArrayExceptSelf
+{
+    public static class ProductArrayExceptSelf
+    {
+        public static int[] ProductExceptSelf(int[] nums)
+        {
+            var product = 1;
+            var numberOfZeroes = 0;
+            var zeroTraverse = false;
+            int[] result = new int[nums.Length];
+
+            for (var i = 0; i < nums.Length; i++)
+            {
+                if (nums[i] != 0)
+                {
+                    product *= nums[i];
+                }
+                else
+                {
+                    zeroTraverse = true;
+                    numberOfZeroes++;
+                }
+            }
+
+            if (numberOfZeroes >= 2)
+            {
+                return result;
+            }
+            else if (zeroTraverse)
+            {
+                for (var i = 0; i < nums.Length; i++)
+                {
+                    if (nums[i] == 0)
+                    {
+                        result[i] = product;
+                    }
+                    else
+                    {
+                        result[i] = 0;
+                    }
+                }
+                return result;
+            }
+
+            for (var i = 0; i < nums.Length; i++)
+            {
+                if (nums[i] != 0)
+                {
+                    result[i] = product / nums[i];
```

Comment:
let's try another way that not using "/"

Reply:
actually I don't know why we should not use "/"

Reply:
interviewer can ask candidates weird anything as a follow up question. Ok, so the target here is not to pass Leetcode's test but to train your skill to be flexible.

Reply:
I have done and pushed that code, please review it, thank you